doc/dev/glossary.md

```diff
@@ -0,0 +1,73 @@
+# Glossary
+
+This page attempts to define and especially disambiguate specialized terms that
+may arise when reading about or developing GREASE.
+
+## Control-flow graphs
+
+"CFG" stands for "control-flow graph". See:
+
+- [The rustc developer guide](https://rustc-dev-guide.rust-lang.org/appendix/background.html#what-is-a-control-flow-graph)
+- [Wikipedia](https://en.wikipedia.org/wiki/Control-flow_graph)
+
+In the context of GREASE, there are a few different kinds of CFGs:
+
+- Macaw CFGs, which arise from code discovery: [`Data.Macaw.CFG.Core`]
+- Crucible CFGs, which can be symbolically executed:
+  - In SSA form: [`Lang.Crucible.CFG.Core`]
+  - In "registerized" form: [`Lang.Crucible.CFG.Reg`]
+
+The `macaw-symbolic` package transforms Macaw CFGs into Crucible CFGs.
+
+Similarly, "expression" or "statement" may refer to expressions or statements in
+any of the above CFGs.
+
+[`Data.Macaw.CFG.Core`]: https://galoisinc.github.io/macaw/Data-Macaw-CFG-Core.html
+[`Lang.Crucible.CFG.Core`]: https://hackage-content.haskell.org/package/crucible/docs/Lang-Crucible-CFG-Core.html
+[`Lang.Crucible.CFG.Reg`]: https://hackage-content.haskell.org/package/crucible/docs/Lang-Crucible-CFG-Reg
+
+## Memory
+
+In the context of GREASE, "memory" may refer to:
+
+- Actual run-time memory (i.e., RAM) of a program under analysis
+- Macaw memory, which is a pre-loader representation of the memory layout
+  extracted from metadata in a binary: [`Data.Macaw.Memory`]
+- The Crucible-LLVM memory model [`Lang.Crucible.LLVM.MemModel`], which serves
+  as the representation of memory during symbolic execution.
+- GREASE's memory model, which is essentially that of Crucible-LLVM mediated by
+  the Macaw Crucible language extension. See [Memory model](../memory-model.md).
+
+[`Data.Macaw.Memory`]: https://galoisinc.github.io/macaw/Data-Macaw-Memory.html
+[`Lang.Crucible.LLVM.MemModel`]: https://hackage-content.haskell.org/package/crucible-llvm/docs/Lang-Crucible-LLVM-MemModel.html
+
+## Common abbreviations and acronyms
```

Comment:
I wonder if GREASE itself should be included in this list of acronyms :)

Reply:
Hopefully you already know what GREASE is if you're on this page!

Reply:
I was thinking more of what the GREASE acronym stands for (this is not explained anywhere else in the docs, as far as I am aware).